Source/JavaScriptCore/wasm/WasmAirIRGenerator.cpp

```diff
@@ -1907,20 +1917,9 @@ inline TypedTmp AirIRGenerator::emitLoadOp(LoadOpType op, ExpressionType pointer
 {
     uint32_t offset = fixupPointerPlusOffset(pointer, uoffset);
 
-    TypedTmp immTmp;
-    TypedTmp newPtr;
     TypedTmp result;
 
-    Arg addrArg;
-    if (Arg::isValidAddrForm(offset, B3::widthForBytes(sizeOfLoadOp(op))))
-        addrArg = Arg::addr(pointer, offset);
-    else {
-        immTmp = g64();
-        newPtr = g64();
-        append(Move, Arg::bigImm(offset), immTmp);
-        append(Add64, immTmp, pointer, newPtr);
-        addrArg = Arg::addr(newPtr);
-    }
+    Arg addrArg = materializeAddrArg(pointer, offset, widthForBytes(sizeOfLoadOp(op)));
```

Comment:
Not strictly RS related, but still a simd-related refactoring that would be nice to get out of the way